src/devices_models/devices/common/objective_function/market_bid.jl

```diff
@@ -9,16 +9,17 @@ function _add_pwl_variables!(
     component_name::String,
     time_period::Int,
     cost_data::PSY.PiecewiseStepData,
-) where {T <: PSY.Component}
-    var_container = lazy_container_addition!(container, PieceWiseLinearBlockOffer(), T)
+    ::Type{U},
+) where {T <: PSY.Component, U <: Union{PieceWiseLinearBlockOffer, PieceWiseLinearBlockDecrementalOffer}}
+    var_container = lazy_container_addition!(container, U(), T)
     # length(PiecewiseStepData) gets number of segments, here we want number of points
     break_points = PSY.get_x_coords(cost_data)
     pwlvars = Array{JuMP.VariableRef}(undef, length(break_points))
     for i in 1:(length(break_points) - 1)
         pwlvars[i] =
             var_container[(component_name, i, time_period)] = JuMP.@variable(
                 get_jump_model(container),
-                base_name = "PieceWiseLinearBlockOffer_$(component_name)_{pwl_$(i), $time_period}",
+                base_name = "$(string(U))_$(component_name)_{pwl_$(i), $time_period}",
```

Comment:
nitpick: `nameof` might be more appropriate here:
```suggestion
                base_name = "$(nameof(U))_$(component_name)_{pwl_$(i), $time_period}",
```